nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -394,14 +393,36 @@ in {
 
         host = lib.mkOption {
           description = lib.mdDoc "Redis host.";
-          type = lib.types.str;
-          default = "127.0.0.1";
+          type = lib.types.nullOr lib.types.str;
+          default = if cfg.redis.createLocally then "127.0.0.1" else null;
+          defaultText = lib.literalExpression ''
+            if config.${opt.redis.createLocally} then "127.0.0.1" else null
+          '';
         };
 
         port = lib.mkOption {
           description = lib.mdDoc "Redis port.";
-          type = lib.types.port;
-          default = 31637;
+          type = lib.types.nullOr lib.types.port;
+          default = if (cfg.redis.createLocally && (!cfg.redis.enableUnixSocket)) then 31637 else null;
+          defaultText = lib.literalExpression ''
+            if (config.${opt.redis.createLocally} && (!config.${opt.redis.enableUnixSocket})) then 31637 else null
+          '';
+        };
+
+        enableUnixSocket = lib.mkOption {
+          description = lib.mdDoc "Use Unix socket";
+          type = lib.types.bool;
+          default = true;
+        };
+
+        nameSpace = lib.mkOption {
```

Comment:
Given that you can have only Mastodon server on a single NixOS system at the moment, this option seems weird and not really useful?

Reply:
Also in the case of local resource creation, the Mastodon redis server is already separate.

Reply:
Before changes redis service, I used a custom database name. After the change, I would not want to reset the redis database for mastodon.

Reply:
So this is a change that only makes sense for you for your specific personal setup? That's not how this works...

Reply:
Ok, removed this option.